### PR TITLE
bluetooth: fast_pair: Set Oberon as default crypto backend

### PR DESCRIPTION
**Pull request by joerchan**
Set Oberon as default crypto backend.
The Mbed TLS APIs are considered as legacy crypto APIs in NCS. The memory usage of the Mbed TLS backend is significantly larger than Oberon or Tinycrypt.

### CODE REVIEW

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/bluetooth_services/services/fast_pair.rst [hunk 1]
```diff
@@ -29,9 +29,8 @@ The following Kconfig options are also available for this module:
 
 * :kconfig:option:`CONFIG_BT_FAST_PAIR_STORAGE_ACCOUNT_KEY_MAX` - The option configures maximum number of stored Account Keys.
 * :kconfig:option:`CONFIG_BT_FAST_PAIR_CRYPTO_TINYCRYPT`, :kconfig:option:`CONFIG_BT_FAST_PAIR_CRYPTO_MBEDTLS`, and :kconfig:option:`CONFIG_BT_FAST_PAIR_CRYPTO_OBERON` - These options are used to select the cryptographic backend for Fast Pair.
-  MbedTLS is used by default, whereas Tinycrypt is used by default for cases of building with TF-M as the Secure Execution Environment (:kconfig:option:`CONFIG_BUILD_WITH_TFM`).
-  This is because in such case the MbedTLS API cannot be directly used by the Fast Pair service.
-  The Oberon backend can be used to limit memory consumption.
+  The Oberon backend is used by default.
+  The Mbed TLS backend is using Mbed TLS crypto APIs which is considered a legacy API.
```
Comment (melwee01) [inline doc/nrf/libraries/bluetooth_services/services/fast_pair.rst:33]:
```suggestion
  The Mbed TLS backend uses the Mbed TLS crypto APIs, which are now considered legacy APIs.
```

Progressive tense to present (is using -> uses), subject verb agreement (APIs vs API, is -> are).

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -416,7 +416,11 @@ Bluetooth libraries and services
       The :c:member:`bt_fast_pair_info_cb.account_key_written` callback can be used to notify the application about the Account Key writes.
     * The :kconfig:option:`CONFIG_BT_FAST_PAIR_STORAGE_USER_RESET_ACTION` Kconfig option to enable a custom user reset action that executes together with the Fast Pair factory reset operation triggered by the :c:func:`bt_fast_pair_factory_reset` function.
 
-  * Updated the salt size in the Fast Pair not discoverable advertising from 1 byte to 2 bytes to align with the Fast Pair specification update.
+  * Updated:
+
+    * Salt size in the Fast Pair not discoverable advertising from 1 byte to 2 bytes to align with the Fast Pair specification update.
```
Comment (melwee01) [inline doc/nrf/releases/release-notes-changelog.rst:421]:
```suggestion
    * Salt size in the Fast Pair not discoverable advertising from 1 byte to 2 bytes, to align with the Fast Pair specification update.
```

**Review by kapi-no — APPROVED** [review]

**Comment by kapi-no** [comment]
@annwoj or @divipillai, please take a look as you typically review documentation for Fast Pair modules

**Review by MarekPieta — APPROVED** [review]

File: subsys/bluetooth/services/fast_pair/Kconfig.fast_pair [hunk 3]
```diff
@@ -58,6 +58,7 @@ config BT_FAST_PAIR_ADVERTISING
 	default y
 	select BT_PERIPHERAL
 	select BT_PRIVACY
+	select ENTROPY_GENERATOR
```
Comment (MarekPieta) [inline subsys/bluetooth/services/fast_pair/Kconfig.fast_pair:61]: Please add the same to `BT_FAST_PAIR_GATT_SERVICE`. It also uses the `sys_csrand_get` API

**Review by MarekPieta — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Review by melwee01 — APPROVED** [review]

**Review by annwoj — APPROVED** [review]
lgtm

**Pull request merged**